File: leetcode/0744-find-smallest-letter-greater-than-target/solution.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int lo = 0, hi = letters.size();
        while (lo < hi) {
            int mid = lo + (hi - lo) / 2;
            
            if (target < letters[mid])
                hi = mid;
            else
                lo = mid + 1;
        }

        return hi == letters.size() ? letters[0] : letters[hi];
    }
};

File: leetcode/0744-find-smallest-letter-greater-than-target/solution.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        lo, hi = 0, len(letters)
        while lo < hi:
            mid = lo + (hi - lo) // 2
            
            if target < letters[mid]:
                hi = mid
            else:
                lo = mid + 1

        return letters[0] if hi == len(letters) else letters[hi]